src/mcelog.c

```diff
@@ -0,0 +1,431 @@
+/*-
+ * collectd - src/mcelog.c
+ *
+ * Copyright(c) 2016 Intel Corporation. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Authors:
+ *   Maryam Tahhan <maryam.tahhan@intel.com>
+ *   Volodymyr Mytnyk <volodymyrx.mytnyk@intel.com>
+ *   Taras Chornyi <tarasx.chornyi@intel.com>
+ */
+
+#include "common.h"
+#include "collectd.h"
+
+#include <poll.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#define MCELOG_PLUGIN "mcelog"
+#define BUFF_SIZE 1024
+#define POLL_TIMEOUT 1000 /* ms */
+
+struct mcelog_config_s {
+  char socket_path[PATH_MAX];   /* mcelog client socket */
+  char logfile[PATH_MAX];       /* mcelog logfile */
+  struct sockaddr_un unix_sock; /* mcelog client socket */
+  pthread_t tid;                /* poll thread id */
+  int sock_fd;                  /* mcelog server socket fd */
+};
+
+struct mcelog_memory_rec_s {
+  char location[DATA_MAX_NAME_LEN];  /* SOCKET x CHANNEL x DIMM x*/
+  char dimm_name[DATA_MAX_NAME_LEN]; /* DMI_NAME "DIMM_F1" */
+  int corrected_err_total;           /* x total*/
+  int corrected_err_timed;           /* x in 24h*/
+  char corrected_err_timed_period[DATA_MAX_NAME_LEN];
+  int uncorrected_err_total; /* x total*/
+  int uncorrected_err_timed; /* x in 24h*/
+  char uncorrected_err_timed_period[DATA_MAX_NAME_LEN];
+};
+
+typedef struct mcelog_config_s mcelog_config_t;
+typedef struct mcelog_memory_rec_s mcelog_memory_rec_t;
+
+mcelog_config_t g_mcelog_config;
```

Comment:
I'd like to provide way to reinitalize   g_mcelog_config struct, so  how do you see usage of compound literals there ? I mean:
```c
static mcelog_config_t g_mcelog_config ;$
// skippded lines
static void mcelog_config_init_default(void) {$
  g_mcelog_config = (mcelog_config_t){$
    .logfile = "/var/log/mcelog",$
    .unix_sock = {$
      .sun_family = AF_UNIX,$
      .sun_path = "/var/run/mcelog-client",$
    },$
    .tid = 0,$
    .sock_fd = -1,$
  };$
```